src/mp2_eri_gpw.F

```diff
@@ -243,8 +245,35 @@ SUBROUTINE mp2_eri_2c_integrate_gpw(qs_env, para_env_sub, dimen_RI, mo_coeff, my
 
          CALL timestop(handle2)
 
-         ! integrate the little bastards
          offset = 0
+         ! Prepare offset ahead of OMP parallel loop
+         max_nseta = 0
+         DO iatom = 1, natom
+            ikind = kind_of(iatom)
+            CALL get_qs_kind(qs_kind=qs_kind_set(ikind), basis_set=basis_set_a, basis_type="RI_AUX")
+            nseta = basis_set_a%nset
+            max_nseta = MAX(nseta, max_nseta)
+         END DO
+         ALLOCATE (offset_2d(max_nseta, natom))
+
+         DO iatom = 1, natom
+            ikind = kind_of(iatom)
+            CALL get_qs_kind(qs_kind=qs_kind_set(ikind), basis_set=basis_set_a, basis_type="RI_AUX")
+            nseta = basis_set_a%nset
+            nsgfa => basis_set_a%nsgf_set
+            DO iset = 1, nseta
+               offset = offset + nsgfa(iset)
+               offset_2d(iset, iatom) = offset
+            END DO
+         END DO
+
+         ! integrate the little bastards
+         !$OMP PARALLEL DO DEFAULT(SHARED) &
```

Comment:
As a precaution we require that every OpenMP parallel region uses `DEFAULT(NONE)`.